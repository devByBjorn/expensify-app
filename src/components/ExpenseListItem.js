// Export a stateless functional component
// description, amount, createdAt
import React from 'react'
import { Link } from 'react-router-dom'
import moment from 'moment'
import numeral from 'numeral'

const ExpenseListItem = ({ amount, createdAt, description, id }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>
      {numeral(amount / 100).format('$,00.00')}
    -
    {moment(createdAt).format('MMMM Do, YYYY')}</p>
  </div>
)

export default ExpenseListItem