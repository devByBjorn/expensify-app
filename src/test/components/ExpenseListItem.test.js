import React from 'react'
import { shallow } from 'enzyme'
import ExpenseListItem from '../../components/ExpenseListItem'
import { expenses, prevState } from '../fixtures/expenses'

test('should render ExpenseLIstItem', () => {
  const wrapper = shallow(<ExpenseListItem {...prevState[0]} />)
  expect(wrapper).toMatchSnapshot()
})