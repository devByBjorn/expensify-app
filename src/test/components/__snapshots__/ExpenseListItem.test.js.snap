// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseLIstItem 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/3633aceb-d0d9-44d8-a8f6-59c0f5489c3e"
  >
    <h3>
      Water bill
    </h3>
  </Link>
  <p>
    4500
     - 
    200
  </p>
</div>
`;
